packages/react/src/SelectPanel/SelectPanel.tsx

```diff
@@ -131,14 +131,17 @@ interface SelectPanelBaseProps {
     text: string | React.ReactElement
     variant: 'info' | 'warning' | 'error'
   }
-  onCancel?: () => void
 }
 
+// onCancel is optional with variant=anchored, but required with variant=modal
+type SelectPanelVariantProps = {variant?: 'anchored'; onCancel?: () => void} | {variant: 'modal'; onCancel: () => void}
```

Comment:
Note for reviewer: Moved the variant + onCancel prop combo here. You can always pass an onCancel but you especially must pass it with variant=modal

Reply:
Tried to be smart, backfired in dotcom of course 😓